Sources/StreamChat/Workers/MessageUpdater.swift

```diff
@@ -465,21 +465,18 @@ class MessageUpdater: Worker {
                 throw ClientError.MessageDoesNotExist(messageId: messageId)
             }
 
-            switch messageDTO.localMessageState {
-            case nil, .pendingSync, .syncingFailed, .deletingFailed:
-                try session.pin(message: messageDTO, pinning: pinning)
-                messageDTO.localMessageState = .pendingSync
```

Comment:
I remember now why this was useful 😅 If the pinning failed, we would be able to retry it since editing a message is retryable. 

We can still use the new approach; the only thing we need to add is that if the request fails, we need to unpin it again.

Reply:
why do we remove this code, can you explain?

Reply:
Noting that depending on the localMessageState we use different mechanisms for making the API request. Pending sync == MessageEditor, pendingSend == MessageSender.
These types use different endpoints.

Reply:
Fair enough, good point 👍

Reply:
@martinmitrevski, this removes the editing flow for pinning a message and uses the same flow as adding a reaction with the new partial update endpoint. This is because the MessageEditor has an observer that sends a message update all the time the local state changes.